test/features/showandhide/basicFields.feature

```diff
@@ -0,0 +1,39 @@
+@functional @showandhide
+Feature: Hide and show of basic fields
+
+  Background:
+    Given I have logged in
+
+  Scenario Outline: fields on the same page visible when show & hide condition is met based on mandatory and optional fields
+    Given a case type containing conditionals exists
+    And I start createCase event
+    When I enter '<value>' into the text field with id '<fieldId>'
+    Then the field with label '<expectedLabel>' is visible
+
+    Examples:
+      | value            | fieldId           | expectedLabel          |
+      | showmethemoney   | TextField         | Text Field 2           |
```

Comment:
if we introduce the step I suggested above, we could reuse that same step here to do the verification. i.e. 
```
Then I expect the page to be in the following state:
| fieldId | value |
```